crates/turbo-trace/src/tracer.rs

```diff
@@ -31,29 +36,52 @@ pub struct Tracer {
     import_type: ImportType,
 }
 
-#[derive(Debug, Error, Diagnostic)]
+#[derive(Clone, Debug, Error, Diagnostic)]
 pub enum TraceError {
     #[error("failed to parse file {}: {:?}", .0, .1)]
     ParseError(AbsoluteSystemPathBuf, swc_ecma_parser::error::Error),
     #[error("failed to read file: {0}")]
     FileNotFound(AbsoluteSystemPathBuf),
     #[error(transparent)]
-    PathEncoding(PathError),
+    PathEncoding(Arc<PathError>),
     #[error("tracing a root file `{0}`, no parent found")]
     RootFile(AbsoluteSystemPathBuf),
-    #[error("failed to resolve import to `{path}`")]
+    #[error("failed to resolve import to `{import}` in `{file_path}`")]
     Resolve {
-        path: String,
+        import: String,
+        file_path: String,
         #[label("import here")]
         span: SourceSpan,
         #[source_code]
-        text: NamedSource,
+        text: String,
     },
     #[error("failed to walk files")]
-    GlobError(#[from] globwalk::WalkError),
+    GlobError(Arc<globwalk::WalkError>),
+}
+
+impl TraceResult {
+    pub fn emit_errors(&self) {
```

Comment:
So do proc macros mess up unused code detection? Could you add a decl to silence that warning?

Reply:
I think it's actually [the combination binary/library crate](https://github.com/rust-lang/rust/issues/95513) that's causing this. Because from the binary's perspective this is unused